### PR TITLE
Bugfix FXIOS-16036 [v119] Pull-to-refresh doesn't hide the loading spinner after the page has reloaded

### PR DESCRIPTION
**Pull request by aalaaeid**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-TODO)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16036)

## :bulb: Description
This fixes Pull-to-refresh doesn't hide the loading spinner after the page has reloaded

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/TabManagement/Tab.swift [hunk 1]
```diff
@@ -660,7 +660,10 @@ class Tab: NSObject {
     @objc
     func reloadPage() {
         reload()
-        self.webView?.scrollView.refreshControl?.endRefreshing()
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(1)) {
```
Comment (adudenamedruby) [inline Client/TabManagement/Tab.swift:664]: Heya. So, I was thinking about this and, just having a delay seems.... it doesn't feel good. Is there something we can do to, for example, maybe observe the webView's loading progress, and call this when it's done loading?

**Comment by aalaaeid** [comment]
WIP

**Comment by iMostfa** [comment]
Thanks a lot <3 
i needed this bug to be fixed since since a while

**Comment by aalaaeid** [comment]
> Thanks a lot <3
> i needed this bug to be fixed since since a while

I'm glad I could finally fix the bug that's been bothering my friend! 🫢
Tlmeza. 🫡

**Review by adudenamedruby — APPROVED** [review]

**Pull request merged**